src/formatDate.js

```diff
@@ -8,7 +8,35 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separatorFrom = fromFormat.pop();
+  const separatorTo = toFormat.pop();
+  const dateArr = date.split(separatorFrom);
+  const resultArr = [];
+
+  for (let i = 0; i < toFormat.length; i++) {
+    const format = toFormat[i];
+    const index = fromFormat.indexOf(format);
+
+    if (index === -1) {
+      if (format === 'YYYY') {
+        const fromIndex = fromFormat.indexOf('YY');
+        const year = dateArr[fromIndex];
+
+        resultArr.push(year < 30 ? '20' + year : '19' + year);
+      } else if (format === 'YY') {
+        const fromIndex = fromFormat.indexOf('YYYY');
+        const year = dateArr[fromIndex];
+
+        resultArr.push(year.slice(-2));
+      }
+    } else {
+      resultArr.push(dateArr[index]);
+    }
```

Comment:
The current implementation assumes that the `fromFormat` and `toFormat` arrays have the same length and the same set of format elements (excluding separators). It's important to validate the formats and handle cases where the formats do not match, as this can lead to incorrect indexing and unexpected results.